### PR TITLE
Use EXTRACT_AND_REMOVE instead of KEEP for sling initial content policy

### PR DESCRIPTION
**Pull request by niekraaijmakers**
Invoking cpconvert with sling initial content - EXTRACT_AND_REMOVE > putting in a separate folder

## Related Issue
https://github.com/adobe/aemanalyser-maven-plugin/issues/172

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by bosschaert** [comment]
@niekraaijmakers Maybe make this a draft PR if it's still work in progress?

**Comment by niekraaijmakers** [comment]
@bosschaert @cziegeler ready for review

**Comment by niekraaijmakers** [comment]
@karlpauls  @rombert please help out :)

File: aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java [hunk 1]
```diff
@@ -128,34 +141,40 @@ public void convert(final Map<String, File> contentPackages) throws IOException,
...
+                            bundlesOutputDir
+                        )
+                    )
+                    .setBundleSlingInitialContentExtractor(bundleSlingInitialContentExtractor)
                     .setEntryHandlersManager(
                         new DefaultEntryHandlersManager(Collections.emptyMap(), true,
-                                SlingInitialContentPolicy.KEEP, ConverterConstants.SYSTEM_USER_REL_PATH_DEFAULT)
+                                SlingInitialContentPolicy.EXTRACT_AND_REMOVE, ConverterConstants.SYSTEM_USER_REL_PATH_DEFAULT)
                         )
+                    .setAclManager(
```
Comment (rombert) [inline aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java:168]: Do you need to set defaults here?

File: aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java [hunk 2]
```diff
@@ -128,34 +141,40 @@ public void convert(final Map<String, File> contentPackages) throws IOException,
...
+                            new DefaultAclManager()
+                            )
+                    .setIndexManager(
+                            new DefaultIndexManager()
```
Comment (rombert) [inline aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java:172]: Do you need to set defaults here?

File: aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java [hunk 3]
```diff
@@ -177,4 +192,33 @@ private ResourceFilter getResourceFilter() {
...
+        public String deploy(ArtifactWriter artifactWriter, String runmode,
+                             ArtifactId id) throws IOException {
+            if (runmode != null) {
+                mutableContentPackagesWithRunMode.put(id, runmode);
+            }
+            return super.deploy(artifactWriter, runmode, id);
+        }
+        
+        public void logMutableContentPackages() {
+            if ( !mutableContentPackagesWithRunMode.isEmpty() ) {
```
Comment (rombert) [inline aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java:215]: You don't need this empty check here.

**Review by rombert — CHANGES_REQUESTED** [review]
Overall LGTM, some minor needs. It would be good to for someone else more familiar with the codebase to review as well.

One things which I would like to ask is to create an issue in _this_ project describing the change. We don't know what this SKYOPS thing is :-)

File: aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java (hunk 2 above)
Comment (niekraaijmakers) [reply aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java:172]: yes

File: aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java (hunk 1 above)
Comment (niekraaijmakers) [reply aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java:168]: yes

**Review by rombert — APPROVED** [review]
LGTM, thanks.

**Review by bosschaert — APPROVED** [review]

**Pull request merged**